### PR TITLE
KOGITO-424: Deprecate public usages of KieRuntimeBuilder

### PR DESCRIPTION
**Pull request by evacchi**
convert examples to the new APIs / remove outdated examples

### CODE REVIEW

**Comment by evacchi** [comment]
I upgraded to units the rules part, and I have adjusted the BPMN to invoke the right URLs but can you guys double check the integration with processes still work? haven't had time yet @MarianMacik @radtriste :pray:

**Comment by radtriste** [comment]
Giving it a try :)

**Comment by radtriste** [comment]
```
2020-04-03 10:49:24,220 INFO  [io.quarkus] (main) onboarding 8.0.0-SNAPSHOT (powered by Quarkus 1.3.0.Final) started in 1.392s. Listening on: http://0.0.0.0:8080
2020-04-03 10:49:24,220 INFO  [io.quarkus] (main) Profile prod activated. 
2020-04-03 10:49:24,221 INFO  [io.quarkus] (main) Installed features: [cdi, kogito, resteasy, resteasy-jackson, smallrye-openapi]
2020-04-03 10:50:28,328 ERROR [org.jbp.wor.ins.imp.WorkflowProcessInstanceImpl] (executor-thread-1) Unexpected error (id 1e3a421e-8dbe-417c-b684-aec7166a298d) while executing node Validate employee in process instance a939aaf3-c73a-4614-b900-559d2af790a5: org.jbpm.workflow.instance.WorkflowRuntimeException: [onboarding.onboarding:a939aaf3-c73a-4614-b900-559d2af790a5 - Validate employee:7] -- Unsuccessful response from service Not Found (code 404)
```

This is the call made on onboarding application
```
      { 
        "employee" : {
          "firstName" : "Mark", 
          "lastName" : "Test", 
          "personalId" : "xxx-yy-zzz", 
          "birthDate" : "1995-12-10T14:50:12.123+02:00", 
          "address" : {
            "country" : "US", 
            "city" : "Boston", 
            "street" : "any street 3", 
            "zipCode" : "10001"
          }
        }
      }
```

**Comment by radtriste** [comment]
did urls changed with your switch to ruleunit ?

**Comment by evacchi** [comment]
@radtriste yes I had to. signatures may have also changed a bit. I though I had committed the changes to the BPM part too, but I suppose I forgot.. there you go https://github.com/kiegroup/kogito-examples/pull/152/files#diff-fade9a0c4c893b0f446c814c394d615c I am only uncertain about the signatures, some test cases were missing I had to guess

**Comment by radtriste** [comment]
did you change the onboarding bpmn files ?
here you just updated the `local` on onboarding side ...
how does the service discovery on onboarding knows which path to call ?

**Comment by evacchi** [comment]
I don't know how that part works :D

File: onboarding-example/hr/README.md [hunk 1]
```diff
@@ -60,15 +75,45 @@ As response the employee details including the new employee id and email address
...
+    "birthDate": "2012-12-10T14:50:12.123+02:00",
+    "address": {
+      "country": "US",
+      "city": "Boston",
+      "street": "any street 3",
+      "zipCode": "10001"
+    }
+  }
+}
+' http://localhost:8081/department/first
```
Comment (radtriste) [inline onboarding-example/hr/README.md:116]: `oc new-app hr-service:latest -l department=process,id=process,employeeValidation=process`
`oc new-app hr-service:latest -l department/first=process,id=process,employee-validation/first=process`

**Review by radtriste — APPROVED** [review]
see my readme comment

File: onboarding-example/hr/README.md [hunk 2]
```diff
@@ -81,8 +126,8 @@ NOTE: Make sure that kogito S2I image builders are available to your OpenShift e
 * Create new binary build and start it by uploading content of the current directory
 
 ```sh
-oc new-build myproject/kogito-quarkus-ubi8-s2i --binary=true --name=hr-service-builder
-oc start-build hr-service-builder --from-dir . --incremental=true
+oc new-app hr-service:latest -l department=process,id=process,employeeValidation=process
```
Comment (radtriste) [inline onboarding-example/hr/README.md]: I meant change it on line #142 sorry. This part was good ;)

Comment (evacchi) [reply onboarding-example/hr/README.md]: https://sadtrombone.com/

Comment (evacchi) [reply onboarding-example/hr/README.md]: well that makes much more sense now that i think about it haha

Comment (radtriste) [reply onboarding-example/hr/README.md]: I hate GH for that. When there is no change on a specific line, you cannot comment on it ...

**Review by radtriste — APPROVED** [review]
lgtm

@MarianMacik can you have a check as well ?

We will need as well to merge that one for the tests to work with master: https://github.com/kiegroup/kogito-cloud-operator/pull/268

**Review by MarianMacik — APPROVED** [review]
Just quickly went through it as I didn't have time to run it, looks good.

**Comment by radtriste** [comment]
@evacchi I think we can merge this one
also do you plan to cherry-pick in 0.9.x ?

**Comment by evacchi** [comment]
@radtriste I do, waiting for https://github.com/kiegroup/kogito-examples/pull/187 to be merged first. ~~AND WHY IS THIS RED AGAIN D:~~

**Pull request merged**